sourcePage.js

```diff
@@ -17,61 +22,177 @@ function getPage() {
         // Run script to get HTML after major has been retrieved from storage
         chrome.runtime.sendMessage({
             action: "getSource",
-            source: DOMtoString(selectedMajor)
+            source: getHtml(selectedMajor)
         });
     });
 }
 
 // Reads through HTML and gets class names, prints to popup with concentration
-function DOMtoString(selectedMajor) {
-    var html = '';
+function getHtml(selectedMajor) {
+    selectedMajor = selectedMajor.slice(1, -1); // Take quotes off string
+    var html = 'Major: ' + selectedMajor + '\n';
     var nodes = document.body.getElementsByClassName('classAbbreviation');
-    selectMajor = selectedMajor.substr(1, selectedMajor.length - 2); // Shave off quotation marks
-    html += ('My major is ' + selectedMajor + '\n');
 
     // Parse html of page
     var i = 0; 
     var l = nodes.length;
     for (i = 0; i < l; i++) {
```

Comment:
move i inside loop. setting it to 0 first is redundant; you can just do var i; instead